cloud/amazon/iam_policy.py

```diff
@@ -183,6 +183,14 @@ def role_action(module, iam, name, policy_name, skip, pdoc, state):
     current_policies = [cp for cp in iam.list_role_policies(name).
                                         list_role_policies_result.
                                         policy_names]
+  except boto.exception.BotoServerError as e:
+    if e.error_code == "NoSuchEntity":
+      # Role doesn't exist so it's safe to assume the policy doesn't either
```

Comment:
If the role doesn't exist then there can be no managed policies attached to it and no inline policies will exist. Also the current version of Boto (2.38.0) can't check for managed policies on a role, although this functionality is currently being added (develop branch of Boto).